internal/slicer/slicer.go

```diff
@@ -397,3 +391,63 @@ func createFile(targetPath string, pathInfo setup.PathInfo) (*fsutil.Entry, erro
 		MakeParents: true,
 	})
 }
+
+// selectPkgArchives selects the appropriate archive for each selected slice
+// package. It returns a map of archives indexed by package names.
+func selectPkgArchives(archives map[string]archive.Archive, selection *setup.Selection) (map[string]archive.Archive, error) {
+	// Sort the archives by their priority value.
+	releaseArchives := make([]*setup.Archive, 0, len(selection.Release.Archives))
+	for _, archive := range selection.Release.Archives {
+		if archive.Priority < 0 {
+			// Ignore negative priority archives unless a package specifically
+			// asks for it with the "archive" field.
+			continue
+		}
+		releaseArchives = append(releaseArchives, archive)
+	}
+	slices.SortFunc(releaseArchives, func(a, b *setup.Archive) int {
+		if a.Priority > b.Priority {
+			return -1
+		}
+		if a.Priority < b.Priority {
+			return 1
+		}
+		return 0
+	})
+
+	// Select the appropriate archive for each slice package.
+	pkgArchives := make(map[string]archive.Archive)
+	for _, s := range selection.Slices {
+		pkg := selection.Release.Packages[s.Package]
+		if _, ok := pkgArchives[pkg.Name]; ok {
+			continue
+		}
+		if pkg.Archive == "" {
+			// If the package has not pinned any archive, choose the highest
+			// priority archive in which the package exists.
+			var chosen archive.Archive
+			for _, releaseArchive := range releaseArchives {
+				archive := archives[releaseArchive.Name]
+				if archive != nil && archive.Exists(pkg.Name) {
+					chosen = archive
+					break
+				}
+			}
+			if chosen == nil {
+				return nil, fmt.Errorf("slice package %q missing from archive(s)", pkg.Name)
```

Comment:
I think we should be consistent with `archive.go` and the two error messages here should be:
```go
		return nil, fmt.Errorf("cannot find package %q in archive", pkg)
```

Reply:
So the previous errors for this (/these) case(s) used to be `slice package %q missing from archive`. I only added `(s)` for the archive-not-pinned situation. Let me know what you think!

Reply:
Upon discussion, let's go ahead with the proposed change. TODO @rebornplusplus.

Reply:
Done in https://github.com/letFunny/chisel/pull/12/commits/1f4a4545e7dfbde0d5309b2a3ed4474a28afd460.